react-accordion/my-app/src/TopicHeader.tsx

```diff
@@ -0,0 +1,20 @@
+type Props = {
+  id: number;
+  currentDisplayId: number | undefined;
+  title: string;
+  onHeaderClick: (id: number | undefined) => void;
+};
+export function TopicHeader({
+  id,
+  currentDisplayId,
+  title,
+  onHeaderClick,
+}: Props) {
+  return (
+    <div
+      onClick={() => onHeaderClick(currentDisplayId === id ? undefined : id)}
```

Comment:
I think you want to process clicks even if the card is currently displayed. If you click the one that is currently displayed, it should close. You should fix this.